invoices/invoiceregistry.go

```diff
@@ -842,11 +855,28 @@ func (i *InvoiceRegistry) NotifyExitHopHtlc(rHash lntypes.Hash,
 
 	switch invoiceHtlc.State {
 	case channeldb.HtlcStateCanceled:
+		// If the invoice is in a cancelled state, we expect the
+		// resolution to be a failure resolution.
+		failureResult, ok := result.(FailureResolutionResult)
```

Comment:
indeed, as discussed offline updated to return resolution result in callback. 
The downside here is that accepts are nil resolutions so I've added logging so that we still get some visibility on accept resolution outcomes. It's a bit hacky, but I would like to eventually follow up with a refactor which introduces an accept resolution so we don't need to do nil checking for accepts.